### ISSUE TITLE
Feature request: Automatically merge history

### ISSUE THREAD

**Issue by mllg** [jethrokuan/fzf#117]
I find it handy to have the history merged automatically to be able to find commands I've typed in concurrent sessions. This would just require to add a single line, but maybe there should be an environment variable to control this?
```diff
diff --git a/functions/__fzf_reverse_isearch.fish b/functions/__fzf_reverse_isearch.fish
index 71cbb09..6069481 100644
--- a/functions/__fzf_reverse_isearch.fish
+++ b/functions/__fzf_reverse_isearch.fish
@@ -1,4 +1,5 @@
 function __fzf_reverse_isearch
+    history --merge
     history -z | eval (__fzfcmd) --read0 --tiebreak=index --toggle-sort=ctrl-r $FZF_DEFAULT_OPTS $FZF_REVERSE_ISEARCH_OPTS -q '(commandline)' | perl -pe 'chomp if eof' | read -lz result
     and commandline -- $result
     commandline -f repaint
```

**Comment by jethrokuan** [comment]
I'm fine with this being default behaviour. I think the command is `history merge` though. Feel free to open a PR.

**Comment by jethrokuan** [comment]
Thanks! Resolved in #118
